src/theory/quantifiers/ieval/state.cpp

```diff
@@ -133,6 +313,171 @@ const PatTermInfo& State::getPatTermInfo(TNode p) const
   return it->second;
 }
 
+void State::notifyPatternEqGround(TNode p, TNode g)
+{
+  Assert(!g.isNull());
+  Assert(!expr::hasBoundVar(g));
+  Assert(d_tec->evaluateBase(*this, g) == g);
```

Comment:
IIUC this is not capturing to when `g` is `some`? My understanding from the docs is that it cannot be `some`, has to either ground or `none`.

Reply:
It can be some, the doc was incomplete, I've updated.